### DISCUSSION TITLE
Error using 'genericjsonapi' in Superset

### DISCUSSION THREAD

**Discussion by sh4m3** [apache/superset#27419] [Q&A / Help]
I'm trying to connect to REST API with shillelagh genericjsonapi adapter and i get this error:

```
Shillelagh Error
shillelagh error: module 'jsonpath' has no attribute 'findall'
```
This in my conf:

SQLALCHEMY URI: shillelagh+safe://

ENGINE PARAMETERS:

{"connect_args":{"adapters":["genericjsonapi"],"adapter_kwargs":{"genericjsonapi":{"request_headers":{"authorization":"XXXXXXX"}}}}}

My query is:

SELECT * FROM "[https://restapiurl/RestAPI/services/articulos2#$.articulo[*]](http://restapiurl:8081/RestAPI/services/articulos2#$.articulo%5B*%5D)"

And service response is OK and servers a JSON like this:

{
"articulo": [
{
"codigo": "01023",
"descripcion": "GALLO 450 gr. MARAVILLA (C18)",
"descripcion_art": "GALLO 450 gr. MARAVILLA (C18)",
"familia_nivel1": "01",
"familia": "0102",
"proveedor": 2069,
"nombre_proveedor": " GALLO SAU",
"ref_proveedor": "100202",
"unid_por_envase": 18.0,
"peso_por_caja_gr": 9.414,
"peso_unitario_gr": 0.523,
"precio1": 0.999,
"minima_venta": 18.0,
"minima_frac_venta": 18.0,
"observaciones1": "Palé 9*8",
"blo_ven": 0,
"imagenruta": "",
"imagen": "",
"setPrecioConsultaDoble": "0.999",
"grupos_dtos_venta": 2,
"claseMaterial": "U",
"stock": 108.0,
}
]
}

Any suggestions?

Thanks!

**Comment by raendurikon** [comment]
I am running into the same Problem. Did you ever fix it?

**Comment by Chinguun0301** [comment]
I think some dependencies are missing for the adapter. try pip install "shillelagh[all]" or pip install apache-superset[genericjsonapi]

**Comment by msyavuz** [comment]
[This](https://github.com/betodealmeida/shillelagh/issues/448) might be related.
